pkg/service/pipeline.go

```diff
@@ -953,6 +954,119 @@ func (s *service) triggerPipeline(
 }
 
+// TODO tillknuesting: consider	to merge with with triggerPipeline
+func (s *service) triggerPipelineWithStream(
+	ctx context.Context,
+	ns resource.Namespace,
+	r *datamodel.Recipe,
+	isAdmin bool,
+	pipelineID string,
+	pipelineUID uuid.UUID,
+	pipelineReleaseID string,
+	pipelineReleaseUID uuid.UUID,
+	pipelineInputs []*structpb.Struct,
+	pipelineSecrets map[string]string,
+	pipelineTriggerID string,
+	returnTraces bool,
+	streamChan chan<- StreamResult) error {
+
+	logger, _ := logger.GetZapLogger(ctx)
+
+	memoryKey, err := s.preTriggerPipeline(ctx, isAdmin, ns, r, pipelineTriggerID, pipelineInputs, pipelineSecrets)
+	if err != nil {
+		return fmt.Errorf("from preTriggerPipeline: %w", err)
+	}
+	defer recipe.Purge(ctx, s.redisClient, pipelineTriggerID)
+
+	workflowOptions := client.StartWorkflowOptions{
+		ID:                       pipelineTriggerID,
+		TaskQueue:                worker.TaskQueue,
+		WorkflowExecutionTimeout: time.Duration(config.Config.Server.Workflow.MaxWorkflowTimeout) * time.Second,
+		RetryPolicy: &temporal.RetryPolicy{
+			MaximumAttempts: config.Config.Server.Workflow.MaxWorkflowRetry,
+		},
+	}
+
+	userUID := uuid.FromStringOrNil(resource.GetRequestSingleHeader(ctx, constant.HeaderUserUIDKey))
+
+	we, err := s.temporalClient.ExecuteWorkflow(
+		ctx,
+		workflowOptions,
+		"TriggerPipelineWorkflow",
+		&worker.TriggerPipelineWorkflowParam{
+			BatchSize:        len(pipelineInputs),
+			MemoryStorageKey: memoryKey,
+			SystemVariables: recipe.SystemVariables{
+				PipelineID:          pipelineID,
+				PipelineUID:         pipelineUID,
+				PipelineReleaseID:   pipelineReleaseID,
+				PipelineReleaseUID:  pipelineReleaseUID,
+				PipelineRecipe:      r,
+				PipelineOwnerType:   ns.NsType,
+				PipelineOwnerUID:    ns.NsUID,
+				PipelineUserUID:     userUID,
+				HeaderAuthorization: resource.GetRequestSingleHeader(ctx, "authorization"),
+			},
+			Mode: mgmtPB.Mode_MODE_SYNC,
+		})
+	if err != nil {
+		logger.Error(fmt.Sprintf("unable to execute workflow: %s", err.Error()))
+		return fmt.Errorf("unable to execute workflow: %w", err)
+	}
+
+	//TODO: refactor logic to get intermediate results
+	c := s.temporalClient
+	go func() {
+		for {
+			// Add a delay to ensure the workflow has started and the query handler is registered
+			time.Sleep(time.Millisecond * 500) //TODO tillknuesting: make this configurable
+
+			// Query the SecondaryWorkflow status
+			queryResult, err := c.QueryWorkflow(context.Background(), "secondary_workflow", "", "status")
+			if err != nil {
+				log.Println("Unable to query workflow", err)
+				continue
+			}
+
+			var status string
+			if err := queryResult.Get(&status); err != nil {
+				log.Println("Unable to get query result", err)
+				continue
+			}
+
+			fmt.Println("SecondaryWorkflow status:", status)
+
+			//TODO tillknuesting: Refactor to load result from memory here and put in channel
+			outputs, metadata, _ := s.getOutputsAndMetadata(ctx, pipelineTriggerID, r, true)
```

Comment:
probably need to refactor getOutputsAndMetadata

Reply:
Agree